### PR TITLE
modify kustomize fixture to installation fixture for all e2e tests

### PR DESCRIPTION
**Pull request by jsitu777**
**Which issue is resolved by this Pull Request:**
Resolves #

**Description of your changes:**
-Modify kustomize fixture to be installation fixture so it can have both kustomize and helm as installation options
-added delete_cluster option in addition to keep_success, cluster will not be deleted unless user pass in --delete_cluster
-minor fix to istio installation path to make istio-system namespace to be deleted the last

**Testing:**
- [ ] Unit tests pass
- [x] e2e tests pass
- Details about new tests (If this PR adds a new feature)
- Details about any manual tests performed

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/e2e/conftest.py [hunk 1]
```diff
@@ -65,6 +88,24 @@ def get_secretkey(request):
         pytest.fail("--secretkey is required")
     return secret_key
 
+def get_installation_option(request):
+    installation_option = request.config.getoption("--installation_option")
+    if not installation_option:
+        installation_option = "kustomize"
+    return installation_option
+
+def get_depployment_option(request):
```
Comment (rrrkharse) [inline tests/e2e/conftest.py:97]: nit:
get_depployment_option --> get_deployment_option

File: tests/e2e/conftest.py [hunk 2]
```diff
@@ -65,6 +88,24 @@ def get_secretkey(request):
...
+    if not deployment_option:
+        deployment_option = "vanilla"
+    return deployment_option
+
+def get_aws_telemetry_option(request):
+    aws_telemetry_option = request.config.getoption("--aws_telemetry")
+    if not aws_telemetry_option:
+        aws_telemetry_option = "enable"
+    return aws_telemetry_option
+
```
Comment (rrrkharse) [inline tests/e2e/conftest.py:108]: why is this needed as a separate option? what testing purpose does it serve?

File: tests/e2e/fixtures/cognito_dependencies.py [hunk 3]
```diff
@@ -94,7 +95,11 @@ def on_create():
 
     def on_delete():
         cfg = metadata.get("cognito_dependencies") or cognito_deps
-        delete_cognito_dependency_resources(cfg)
+        print("keep Success Option for Cognito Fixture:")
+        print(keep_successfully_created_resource(request))
+        if keep_successfully_created_resource(request) == False:
+            print("start to uninstall Cognito Dependencies...")
+            delete_cognito_dependency_resources(cfg)
```
Comment (rrrkharse) [inline tests/e2e/fixtures/cognito_dependencies.py:102]: this logic is not necessary, the keep success condition is already handled by `configure_resource_fixture()`

```
def configure_resource_fixture(
    metadata, request, resource_details, metadata_key, on_create, on_delete
):
    """
    Helper method to create resources if required and configure them for teardown.

    If the resource is not present in the metadata a new resource will be created and added to
    the metadata.

    If a resource is present in the metadata it will not be created.

    If the `--keepsuccess` flag is specified, successfully created resources (e.g. those that
    did not raise an exception) will not be deleted.
    """
    successful_creation = False

    def delete():
        if successful_creation and keep_successfully_created_resource(request):
            return
        on_delete()
```

File: tests/e2e/fixtures/installation.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    After all tests are run, uninstalls kubeflow using the manifest at `kustomize_path`
+    if the flag `--keepsuccess` was not provided as a pytest argument.
+    """
+
+    def on_create():
+        install_kubeflow(installation_option, aws_telemetry_option, deployment_option, cluster)
+
+    def on_delete():
+        if keep_successfully_created_resource(request) == False:
```
Comment (rrrkharse) [inline tests/e2e/fixtures/installation.py:55]: not necessary, see above

File: tests/e2e/fixtures/cognito_dependencies.py [hunk 5]
```diff
@@ -118,12 +123,14 @@ def callback():
 
 @pytest.fixture(scope="class")
 def post_deployment_dns_update(
-    metadata, region, request, cluster, cognito_bootstrap, kustomize
+    metadata, region, request, cluster, cognito_bootstrap, installation
 ):
-
+    print ("get into post deployment...")
```
Comment (surajkota) [inline tests/e2e/fixtures/cognito_dependencies.py:128]: nit: remove

File: tests/e2e/utils/kubeflow_installation.py [hunk 6]
```diff
@@ -137,15 +141,19 @@ def build_component(
...
+        print(f"All {component_name} pods are running!")
+   
+@retry (stop_max_attempt_number=3)
+def validate_component_installation(path_dic,component_name):
+    identifier = path_dic[component_name]["validations"]["identifier"]
+    common_label = path_dic[component_name]["validations"]["common_label"]
+    namespace = path_dic[component_name]["validations"]["namespace"]
+    print(f"Waiting for {component_name} pods to be ready ...")
+    retcode = kubectl_wait_pods(common_label, namespace, identifier)
+    assert retcode == 0
```
Comment (rrrkharse) [inline tests/e2e/utils/kubeflow_installation.py:156]: what is the retry interval? any reason not to use `wait_for`?

File: tests/e2e/resources/installation_config/rds-only.yaml [hunk 7]
```diff
@@ -57,7 +57,6 @@ kubeflow-namespace:
     kustomize: 
       - "../../upstream/common/kubeflow-namespace/base"
     helm: "../../charts/common/kubeflow-namespace"
-  namespace: 'kubeflow'
```
Comment (surajkota) [inline tests/e2e/resources/installation_config/rds-only.yaml:60]: why?

File: tests/e2e/tests/test_profile_irsa.py [hunk 8]
```diff
@@ -20,6 +20,7 @@
     get_s3_client,
     get_eks_client,
     wait_for_kfp_run_succeeded_from_run_id,
+    apply_kustomize
```
Comment (surajkota) [inline tests/e2e/tests/test_profile_irsa.py:23]: nit: unused?

**Review by rrrkharse — COMMENTED** [review]
minor comments, thanks for this refactor

File: tests/e2e/tests/test_storage_efs.py [hunk 9]
```diff
@@ -148,13 +149,13 @@ def test_pvc_with_volume(
         subprocess.run(f"kubectl delete pod -n {DEFAULT_USER_NAMESPACE} {write_pod_name}".split())
         subprocess.run(f"kubectl delete pod -n {DEFAULT_USER_NAMESPACE} {read_pod_name}".split())
 
-
+"""
```
Comment (surajkota) [inline tests/e2e/tests/test_storage_efs.py:152]: why is this commented out?

File: tests/e2e/conftest.py (hunk 1 above)
Comment (jsitu777) [reply tests/e2e/conftest.py]: fixed.

File: tests/e2e/fixtures/cognito_dependencies.py [hunk 10]
```diff
@@ -94,7 +95,11 @@ def on_create():
 
     def on_delete():
         cfg = metadata.get("cognito_dependencies") or cognito_deps
-        delete_cognito_dependency_resources(cfg)
+        print("keep Success Option for Cognito Fixture:")
+        print(keep_successfully_created_resource(request))
+        if keep_successfully_created_resource(request) == False:
```
Comment (surajkota) [inline tests/e2e/fixtures/cognito_dependencies.py:100]: why? is keepsuccess is false, resources should be deleted

**Review by surajkota — CHANGES_REQUESTED** [review]

File: tests/e2e/utils/kubeflow_installation.py (hunk 6 above)
Comment (jsitu777) [reply tests/e2e/utils/kubeflow_installation.py:156]: this is for resolve the pod delay issue @surajkota run into on Friday

File: tests/e2e/resources/installation_config/rds-only.yaml (hunk 7 above)
Comment (jsitu777) [reply tests/e2e/resources/installation_config/rds-only.yaml:60]: the namespace is for validation purpose only, we are not validating the kubeflow namespace/pod

File: tests/e2e/tests/test_profile_irsa.py (hunk 8 above)
Comment (jsitu777) [reply tests/e2e/tests/test_profile_irsa.py:23]: removed

File: tests/e2e/tests/test_storage_efs.py (hunk 9 above)
Comment (jsitu777) [reply tests/e2e/tests/test_storage_efs.py:152]: let me uncomment it out again. Basically Meghna combine the dynamic test and static test together and it requires two clusters, so I commented out one test and run it one by one. We should decouple this test in the future.

File: tests/e2e/fixtures/cognito_dependencies.py (hunk 10 above)
Comment (jsitu777) [reply tests/e2e/fixtures/cognito_dependencies.py:100]: no. you delete resources if the keepsuccess flag is not there, which mean it's false
but Rahul mentioned this logic is not necessary. 
```
parser.addoption(
        "--keepsuccess",
        action="store_true",
        default=False,
        help="Keep successfully created resources on delete.",
    )
```
https://github.com/awslabs/kubeflow-manifests/pull/369#discussion_r974895154

File: tests/e2e/conftest.py (hunk 2 above)
Comment (jsitu777) [reply tests/e2e/conftest.py:108]: I think you are right, it's needed for the kubeflow installation script, but we can remove it in the test

File: tests/e2e/fixtures/cognito_dependencies.py (hunk 5 above)
Comment (jsitu777) [reply tests/e2e/fixtures/cognito_dependencies.py:128]: removed it. put there for debugging purpose

File: tests/e2e/utils/kubeflow_installation.py (hunk 6 above)
Comment (rrrkharse) [reply tests/e2e/utils/kubeflow_installation.py:156]: ok

**Comment by jsitu777** [comment]
@surajkota @rrrkharse adjusted the comments in the latest commit.

**Review by rrrkharse — COMMENTED** [review]
When merging can you also make sure that the git commit message doesnt include the test output since it is very large

File: tests/e2e/utils/kubeflow_installation.py (hunk 6 above)
Comment (surajkota) [reply tests/e2e/utils/kubeflow_installation.py:156]: @jsitu777, @rrrkharse is right, we should retry with a delay and try for max of 3 times before failing

ok to change this later

**Review by rrrkharse — APPROVED** [review]
Leaving approval for addressing my comments

File: tests/e2e/utils/kubeflow_installation.py (hunk 6 above)
Comment (surajkota) [reply tests/e2e/utils/kubeflow_installation.py:156]: actually, its really simple with the libary
```
@retry(stop_max_attempt_number=3, wait_fixed=3000)
```

lets add this

**Review by surajkota — COMMENTED** [review]
One comment for adding timed wait between retries, please add and merge

also make sure to not use the default commit message while merging because PR description is reallyy long

**Review by surajkota — APPROVED** [review]
Approving, it can be added later as well
